### ISSUE TITLE
Repl panic on non-zero exit

### ISSUE THREAD

**Issue by maxsnew** [zydeco-lang/zydeco#41]
Reproduce: run `zydeco repl`
Behavior:
```
> ! exit 0
exited with code 0
> ! exit 1
thread 'main' panicked at 'variable does not exist', zydeco-lang/src/dynamics/eval.rs:43:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Expected: should say `exited with code 1`.

**Comment by maxsnew** [comment]
Would be good also to have some way to add a regression test here!

**Comment by LighghtEeloo** [comment]
Addressed in 79a4110.

The cheapest regression test I can think of is feeding `*.in` files and `diff` them against `*.ans` files.
